### PR TITLE
Lurze cyborg bug fix pass 1

### PR DESCRIPTION
**Pull request by lurzefromgermany**
CHANGE LOG

CHANGE 1, Fixed self-repair, so that it is no longer stuck in a loop

CHANGE 2 Added traits to the surgical processor so that medical robots are granted sci-related traits when

CHANGE 3, Changed the readable name on the surgical processor to something fallout-related, Changed flavor text very slightly

Change 4, added flags to robot modules in robot_upgrades.dm So that they appear in the correct list in the exo fab

CHANGE 5, added else if(U in typesof(/obj/item/borg/upgrade/modkit))
				sub_category += "Mining" to mech_fabricator.DM so that mining_designs.dm is placed properly in the exofab screen.

TLDR; Self-repair works, Surgical processor renamed, and adds traits to Medical borgs when installed, allowing them to do chem, research, and surgery. Correctly organized the exofab subcategories

### CODE REVIEW

File: code/game/mecha/mech_fabricator.dm [hunk 1]
```diff
@@ -147,6 +147,8 @@
 					sub_category += "Medical"
 				if(module_types & BORG_MODULE_ENGINEERING)
 					sub_category += "Engineering"
+			else if(U in typesof(/obj/item/borg/upgrade/modkit)) //added else if statment for cyborg items in mining_designs.dm
```
Comment (out-of-phaze) [inline code/game/mecha/mech_fabricator.dm:150]: Use ispath, not typesof. Also, why not just give modkits `module_types = BORG_MODULE_MINER`?

**Review by out-of-phaze — CHANGES_REQUESTED** [review]
Please change line 135 in mech_fabricator.dm to be `if(ispath(built_item, /obj/item/borg/upgrade))`, in addition to the other changes I requested.

File: code/game/mecha/mech_fabricator.dm (hunk 1 above)
Comment (lurzefromgermany) [reply code/game/mecha/mech_fabricator.dm:150]: because it is undefined, and if you try to define it, it just doesn't work. "module_types = " works in robot_upgrades.dm and does not work in mining.dm, that line of code is there, because mining.dm did not want to accept the var

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 2]
```diff
@@ -326,16 +327,15 @@ as performing this in action() will cause the upgrade to end up in the borg inst
...
-		if(U)
-			to_chat(user, "<span class='warning'>This unit is already equipped with a self-repair module.</span>")
-			return FALSE
-
-		icon_state = "selfrepair_off"
-		toggle_action = new /datum/action/item_action/toggle(src)
-		toggle_action.Grant(R)
+    . = ..()
+    if(.)
+        for(var/obj/item/borg/upgrade/U in R.upgrades)
```
[the remark this answers is not in the document]
Comment (lurzefromgermany) [reply code/game/objects/items/robot/robot_upgrades.dm:332]: Last format wasn't working at all, that's why I redid it, I will try your method now. Also, it is adding to the contents. beforehand it was acting as if it was already there when it was not.

[the remark this answers is not in the document]
Comment (lurzefromgermany) [reply code/game/objects/items/robot/robot_upgrades.dm:332]: Yup your suggestion worked, put too much thought into this one.

File: code/game/mecha/mech_fabricator.dm (hunk 1 above)
Comment (out-of-phaze) [reply code/game/mecha/mech_fabricator.dm:150]: `module_flags`

Comment (lurzefromgermany) [reply code/game/mecha/mech_fabricator.dm:150]: Yes, same statement I was making, but with the correct flag. What I meant. If you tell me how I can get mining.dm to accept the var definition, that can be done, It is what I was doing originally, but couldn't get it to work.

Comment (lurzefromgermany) [reply code/game/mecha/mech_fabricator.dm]: Changed, removed code, and added module_types = BORG_MODULE_MINER to line 280 in the proper DM

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 3]
```diff
@@ -504,6 +505,12 @@ as performing this in action() will cause the upgrade to end up in the borg inst
 		R.module.basic_modules += SP
 		R.module.add_module(SP, FALSE, TRUE)
 
+		ADD_TRAIT(R, TRAIT_CHEMWHIZ, src) // Added traits to medical proccessor so that Medical borgs can do medical stuff
+		ADD_TRAIT(R, TRAIT_RESEARCHER, src)
+		ADD_TRAIT(R, TRAIT_MEDICALEXPERT, src)
+		ADD_TRAIT(R, TRAIT_SURGERY_HIGH, src)
+		ADD_TRAIT(R, TRAIT_RESEARCHER, src) // Why is this in here twice? Saw it in followers.dm and didn't want to break it. That's why
```
[the remark this answers is not in the document]
Comment (lurzefromgermany) [reply code/game/objects/items/robot/robot_upgrades.dm]: Removed redundant line

File: code/game/objects/items/robot/robot_upgrades.dm (hunk 2 above)
[the remark this answers is not in the document]
Comment (lurzefromgermany) [reply code/game/objects/items/robot/robot_upgrades.dm]: Reverted to original, added .upgrades

**Review by out-of-phaze — CHANGES_REQUESTED** [review]
Alnost there, just a few changes left.

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 4]
```diff
@@ -504,6 +506,11 @@ as performing this in action() will cause the upgrade to end up in the borg inst
 		R.module.basic_modules += SP
 		R.module.add_module(SP, FALSE, TRUE)
 
+		ADD_TRAIT(R, TRAIT_CHEMWHIZ, src) // Added traits to medical proccessor so that Medical borgs can do medical stuff
```
[the remark this answers is not in the document]
Comment (lurzefromgermany) [reply code/game/objects/items/robot/robot_upgrades.dm:509]: Done so

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 5]
```diff
@@ -485,10 +487,10 @@ as performing this in action() will cause the upgrade to end up in the borg inst
 		R.module.remove_module(S, TRUE)
 
 /obj/item/borg/upgrade/processor
-	name = "medical cyborg surgical processor"
+	name = "General Atomics International surgical processor" //changed name to fit in the theme
```
[the remark this answers is not in the document]
Comment (lurzefromgermany) [reply code/game/objects/items/robot/robot_upgrades.dm:490]: Edited

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 6]
```diff
@@ -504,6 +506,11 @@ as performing this in action() will cause the upgrade to end up in the borg inst
 		R.module.basic_modules += SP
 		R.module.add_module(SP, FALSE, TRUE)
 
+		ADD_TRAIT(R, TRAIT_CHEMWHIZ, CYBORG_ITEM_TRAIT) // Added traits to medical proccessor so that Medical borgs can do medical stuff
+		ADD_TRAIT(R, TRAIT_RESEARCHER, CYBORG_ITEM_TRAIT)
+		ADD_TRAIT(R, TRAIT_MEDICALEXPERT, CYBORG_ITEM_TRAIT)
+		ADD_TRAIT(R, TRAIT_SURGERY_HIGH, CYBORG_ITEM_TRAIT)
+
 /obj/item/borg/upgrade/processor/deactivate(mob/living/silicon/robot/R, user = usr)
```
Comment (out-of-phaze) [inline code/game/objects/items/robot/robot_upgrades.dm:514]: You should probably make this method remove the trait as well.

**Review by out-of-phaze — CHANGES_REQUESTED** [review]
Sorry, didn't realize the surgical processor had a deactivate proc. You'll need to edit that too.

File: code/game/objects/items/robot/robot_upgrades.dm (hunk 6 above)
Comment (lurzefromgermany) [reply code/game/objects/items/robot/robot_upgrades.dm:514]: Done so, Tested, and works.

**Review by out-of-phaze — APPROVED** [review]
Looks good.
